python/cugraph-dgl/examples/muti_trainer_MG_example/workflow.py

```diff
@@ -0,0 +1,233 @@
+
+
+def run_workflow(rank, devices, scheduler_address):
+    # Below sets gpu_num
+    dev_id = devices[rank]
+    initalize_pytorch_worker(dev_id)
+    device = torch.device(f"cuda:{dev_id}")
+    # cugraph dask client initialization
+    client = create_dask_client(scheduler_address)
+
+    # Pytorch training worker initialization
+    dist_init_method = "tcp://{master_ip}:{master_port}".format(
+        master_ip="127.0.0.1", master_port="12346"
+    )
+
+    torch.distributed.init_process_group(
+        backend="nccl",
+        init_method=dist_init_method,
+        world_size=len(devices),
+        rank=rank,
+    )
+
+    print(f"rank {rank}.", flush=True)
+    print("Initalized across GPUs.")
+
+    event = Dask_Event("cugraph_gs_creation_event")
+    if rank == 0:
+        (
+            gs,
+            train_idx,
+            valid_idx,
+            test_idx,
+        ) = create_cugraph_graphstore_from_dgl_dataset(
+            "ogbn-products", single_gpu=False
+        )
+        client.publish_dataset(cugraph_gs=gs)
+        client.publish_dataset(train_idx=train_idx)
+        client.publish_dataset(valid_idx=valid_idx)
+        client.publish_dataset(test_idx=test_idx)
+        event.set()
+    else:
+        if event.wait(timeout=1000):
+            gs = client.get_dataset("cugraph_gs")
+            train_idx = client.get_dataset("train_idx")
+            valid_idx = client.get_dataset("valid_idx")
+            test_idx = client.get_dataset("test_idx")
+        else:
+            raise RuntimeError(f"Fetch cugraph_gs to worker_id {rank} failed")
+
+    torch.distributed.barrier()
+    print(f"Loading cugraph_store to worker {rank} is complete", flush=True)
+    dataloader = create_dataloader(gs, train_idx, device)
+    print("Data Loading Complete", flush=True)
+    del gs  # Clean up gs reference
+    # Comment below
+    st = time.time()
+    for step, (input_nodes, seeds, blocks) in enumerate(dataloader):
+        pass
+    et = time.time()
+    print(f"Warmup loading took = {et-st} s on worker = {rank}")
+    torch.distributed.barrier()
+
+    n_epochs = 30
+    total_st = time.time()
+    for i in range(0, n_epochs):
+        st = time.time()
+        for step, (input_nodes, seeds, blocks) in enumerate(dataloader):
+            pass
+            # print(len(input_nodes))
+            # print(len(seeds))
+            # train_model()
+        et = time.time()
+        print(f"Data Loading took = {et-st} s for epoch = {i} on worker = {rank}")
+    torch.distributed.barrier()
+    total_et = time.time()
+    print(
+        f"Total time taken on n_epochs {n_epochs} = {total_et-total_st} s",
+        f"measured by worker = {rank}",
+    )
+
+    # cleanup dask cluster
+    if rank == 0:
+        client.unpublish_dataset("cugraph_gs")
+        client.unpublish_dataset("train_idx")
+        client.unpublish_dataset("valid_idx")
+        client.unpublish_dataset("test_idx")
+        event.clear()
+
+
+if __name__ == "__main__":
```

Comment:
Any way we could do this just with 2 GPUs, since that cover more user environments?  Like my PyG example, I think it's ok for the GPUs to overlap (although since you're using rmm pool that could make that more complicated).

Reply:
fine with me